pydarn/plotting/rtp.py

```diff
@@ -0,0 +1,438 @@
+            that is does not equal the value.
+
+
+        Raises
+        ------
+        RTPUnknownParameterError
+        RTPInocrrectParameterPlotError
+        RTPNoDataFoundError
+
+        Returns
+        -------
+        im: matplotlib.pyplot.pcolormesh
+            matplotlib object from pcolormesh
+        cb: matplotlib.colorbar
+            matplotlib color bar
+        cmap: matplotlib.cm
+            matplotlib color map object
+        """
+        # Settings
+        settings = {'ground_scatter': False,
+                    'channel': 'all',
+                    'date_fmt': "%y/%m/%d\n%H:%M",
+                    'color_bar': True,
+                    'color_bar_label': '',
+                    'color_map': 'jet',
+                    'boundary': None,
+                    'min_array_filter': dict(),
+                    'max_array_filter': dict(),
+                    'min_scalar_filter': dict(),
+                    'max_scalar_filter': dict(),
+                    'equal_scalar_filter': dict()}
+
+        settings.update(kwargs)
+        beam_num = beam_num
+
+        # If an axes object is not passed in then store
+        # the equivalent object in matplotlib. This allows
+        # for variant matplotlib plotting styles.
+
+        if not ax:
+            ax = plt.gca()
+
+        # Determine if a DmapRecord was passed in, instead of a list
+        try:
+            if isinstance(dmap_data[0][parameter], DmapArray) or\
+               isinstance(dmap_data[0][parameter], DmapScalar):
+                dmap_data = dmap2dict(dmap_data)
+        except KeyError:
+            raise rtp_exceptions.RTPUnknownParameter(parameter)
+        cls.dmap_data = dmap_data
+        cls.__check_data_type(parameter, 'array')
+        # Calculate the time span range, either passed in or calculated
+        # based on the dmap_data records
+        # TODO: move this to it's own method
+        try:
+            if len(kwargs['time_span']) == 2:
+                start_time = kwargs['time_span'][0]
+                end_time = kwargs['time_span'][1]
+                cls.interval_time = None
+            elif len(kwargs['time_span']) == 3:
+                start_time = kwargs['time_span'][0]
+                end_time = kwargs['time_span'][2]
+                cls.interval_time = kwargs['time_span'][1]
+            else:
+                raise IndexError("time_span list must be length of 2 or 3")
+
+        except KeyError:
+            start_time = cls.__time2datetime(cls.dmap_data[0])
+            end_time = cls.__time2datetime(cls.dmap_data[-1])
+
+        # y-axis coordinates, i.e., range gates,
+        # TODO: implement variant other coordinate systems for the y-axis
+        y = np.linspace(0, cls.dmap_data[0]['nrang'],
+                        cls.dmap_data[0]['nrang'])
+        y_max = cls.dmap_data[0]['nrang']
+
+        # z: parameter data mapped into the color mesh
+        z = np.zeros((1, y_max)) * np.nan
```

Comment:
I think it's clearer to use `np.full((1, y_max), fill_value=np.nan)` here and elsewhere